File: spring-web-java/src/main/java/spring/web/java/presentation/converter/DeliveryStatusConverter.java
package spring.web.java.presentation.converter;

import org.springframework.core.convert.converter.Converter;
import org.springframework.stereotype.Component;

import spring.web.java.domain.model.enums.DeliveryStatus;

@Component
public class DeliveryStatusConverter implements Converter<String, DeliveryStatus> {

	@Override
	public DeliveryStatus convert(String source) {
		return DeliveryStatus.of(source);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/converter/DeliveryStatusConverter.kt
package spring.web.kotlin.presentation.converter

import org.springframework.core.convert.converter.Converter
import org.springframework.stereotype.Component
import spring.web.kotlin.domain.model.enums.DeliveryStatus

@Component
class DeliveryStatusConverter : Converter<String, DeliveryStatus> {
    override fun convert(source: String) = DeliveryStatus.of(source)
}